In macOS
```
npm error gyp info spawn args '-Dnode_gyp_dir=/nix/store/k1wbwbv4ay5p0dfkmm2qz3chg3c23f42-nodejs-20.19.5/lib/node_modules/npm/node_modules/node-gyp',
npm error gyp info spawn args '-Dnode_lib_file=/nix/store/qjvcnlnxp7d2grgxwwl6z95ysafc5xq3-nodejs-20.19.5-source/$(Configuration)/node.lib',
npm error gyp info spawn args '-Dmodule_root_dir=/private/tmp/nix-build-vsce-3.6.2.drv-0/source/node_modules/keytar',
npm error gyp info spawn args '-Dnode_engine=v8',
npm error gyp info spawn args '--depth=.',
npm error gyp info spawn args '--no-parallel',
npm error gyp info spawn args '--generator-output',
npm error gyp info spawn args 'build',
npm error gyp info spawn args '-Goutput_dir=.'
npm error gyp info spawn args ]
npm error gyp info spawn make
npm error gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
npm error In file included from ../src/async.cc:4:
npm error ../../node-addon-api/napi.h:1147:39: error: in-class initializer for static data member is not a constant expression
npm error  1147 |     static const napi_typedarray_type unknown_array_type = static_cast<napi_typedarray_type>(-1);
npm error       |                                       ^                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
npm error ../../node-addon-api/napi.h:1147:60: note: integer value -1 is outside the valid range of values [0, 15] for the enumeration type 'napi_typedarray_type'
npm error  1147 |     static const napi_typedarray_type unknown_array_type = static_cast<napi_typedarray_type>(-1);
npm error       |                                                            ^
npm error 1 error generated.
npm error make: *** [keytar.target.mk:126：Release/obj.target/keytar/src/async.o]
```

So I change `keytar` to `@napi-rs/keyring`

--- a/pkgs/applications/editors/vscode/extensions/Google.gemini-cli-vscode-ide-companion/default.nix
+++ b/pkgs/applications/editors/vscode/extensions/Google.gemini-cli-vscode-ide-companion/default.nix
@@ -2,7 +2,6 @@
   lib,
   vscode-utils,
   gemini-cli,
-  vsce,
 }:
 vscode-utils.buildVscodeExtension (finalAttrs: {
   pname = "gemini-cli-vscode-ide-companion";
@@ -14,8 +13,6 @@ vscode-utils.buildVscodeExtension (finalAttrs: {
 
   src = gemini-cli.overrideAttrs (oldAttrs: {
     pname = "gemini-cli-vscode-ide-companion-vsix";
-
-    nativeBuildInputs = gemini-cli.nativeBuildInputs ++ [ vsce ];
 
     installPhase = ''
       runHook preInstall

--- a/pkgs/by-name/ge/gemini-cli/package.nix
+++ b/pkgs/by-name/ge/gemini-cli/package.nix
@@ -1,12 +1,14 @@
 {
   lib,
+  stdenv,
   buildNpmPackage,
   fetchFromGitHub,
-  nix-update-script,
-  ripgrep,
   jq,
   pkg-config,
+  clang_20,
   libsecret,
+  ripgrep,
+  nix-update-script,
 }:
 
 buildNpmPackage (finalAttrs: {
@@ -25,7 +27,8 @@ buildNpmPackage (finalAttrs: {
   nativeBuildInputs = [
     jq
     pkg-config
-  ];
+  ]
+  ++ lib.optionals stdenv.isDarwin [ clang_20 ]; # clang_21 breaks @vscode/vsce's optionalDependencies keytar
 
   buildInputs = [
     ripgrep

--- a/pkgs/by-name/vs/vsce/package.nix
+++ b/pkgs/by-name/vs/vsce/package.nix
@@ -1,11 +1,13 @@
 {
   lib,
+  stdenv,
   buildNpmPackage,
   fetchFromGitHub,
   pkg-config,
   libsecret,
-  python3,
-  testers,
+  nodejs,
+  clang_20,
+  versionCheckHook,
   nix-update-script,
 }:
 
@@ -28,18 +30,19 @@ buildNpmPackage (finalAttrs: {
 
   nativeBuildInputs = [
     pkg-config
-    python3
-  ];
+    nodejs.python
+  ]
+  ++ lib.optionals stdenv.isDarwin [ clang_20 ]; # clang_21 breaks @vscode/vsce's optional dependency keytar
 
   buildInputs = [ libsecret ];
 
   makeCacheWritable = true;
-  npmFlags = [ "--legacy-peer-deps" ];
+
+  doInstallCheck = true;
+  nativeInstallCheckInputs = [ versionCheckHook ];
+  versionCheckProgramArg = "--version";
 
   passthru = {
-    tests.version = testers.testVersion {
-      package = finalAttrs.finalPackage;
-    };
     updateScript = nix-update-script {
       extraArgs = [
         "--version-regex"
@@ -51,7 +54,10 @@ buildNpmPackage (finalAttrs: {
   meta = {
     homepage = "https://github.com/microsoft/vscode-vsce";
     description = "Visual Studio Code Extension Manager";
-    maintainers = with lib.maintainers; [ aaronjheng ];
+    maintainers = with lib.maintainers; [
+      aaronjheng
+      xiaoxiangmoe
+    ];
     license = lib.licenses.mit;
     mainProgram = "vsce";
   };

